src/formula.jl

```diff
@@ -176,8 +176,8 @@ evt(a) = Any[a]
 function Terms(f::Formula)
     rhs = condense(distribute(dospecials(f.rhs)))
     tt = unique(getterms(rhs))
-    tt = tt[!(tt .== 1)]             # drop any explicit 1's
-    noint = (tt .== 0) | (tt .== -1) # should also handle :(-(expr,1))
+    filter!(t -> t != 1, tt)                          # drop any explicit 1's
+    noint = BitArray(map(t -> t == 0 || t == -1, tt)) # should also handle :(-(expr,1))
```

Comment:
Ah. So `broadcast` (but not `map`) will figure out that the return array should be `BitArray`.